### PR TITLE
[NOT READY] Bloodsuckers

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

Ports over the Citadel version of Bloodsuckers (which is a port from our version of OUR version of bloodsuckers, but with code improvements and general other fixes) while bringing back most of the unique fulp features, and also fixing some commented out and broken code to be put back in. 

Help is welcome and appreciated, I've already spent a long time getting to where I am now and I believe I'm just about halfway done.

THINGS TO MAKE SURE WORK:
- [x] Teleport to the coffin and sleep through Torpor
- [x] Gain powers with text and gaining powers/levelling up
- [x] Blood/Level/Sol huds for Bloodsuckers (+Refactor)
- [ ] Antag huds for Bloodsuckers/Vassals (This is currently extremely broken, causing everyone to be named ' !' with the inability to right click on them, and all huds broke down. Please help)
- [x] Crafting menu/learning new things when a coffin is claimed
- [x] Brawn breaking doors open
- [ ] Brawn breaking straighjacket
- [ ] Passively healing Brute damage
- [ ] Healing everything when inside your coffin
- [x] Vassals/Bloodsuckers/Randoms examining the Candelabrum and persuasion racks
- [x] Vassals learning Blahsucker on conversion, losing on deconversion
- [x] Remove the hardlimit on vassals
- [x] Being unvassalized when mindshielded
- [ ] Fix feed so they dont always spray blood everywhere
- [x] Dont get stunned after using Lunge/Haste
- [x] Fix Blahsucker language/icons
- [x] Unable to ride vehicles while using fortitude
- [ ] Ability to drink from a bloodbag
- [ ] Uncomment bloodsucker integration
- [ ] Bloodsucker turning into a Slimeperson doesnt gib them anymore, nor give them slimeperson powers
- [ ] Bloodsucker getting a holoparasite actually has timestop

## Why It's Good For The Game

Enrico...
Requested in #3 

## Changelog
:cl: JohnWillard/SgtHunk
add: Bloodsuckers
/:cl:

### CODE REVIEW

**Comment by JohnFulpWillard** [comment]
While I fixed a lot of the problems for bloodsuckers, I havent been able to fix the few remaining problems with it, and I just don't believe I'm experienced enough to do this. I'll close this PR, I'll re-open it if I manage to figure it out, and everyone is free to take it where I left off

**Comment by JohnFulpWillard** [comment]
re-opening so I can properly merge with master when I need to, to maintain it until someone takes over for me.

**Comment by remove32** [comment]
converting this to draft since you're actively working on it :)

File: fulp_modules/modular_bloodsuckers/bloodsucker/datum_bloodsucker.dm [hunk 1]
```diff
@@ -0,0 +1,742 @@
...
+
+
+// FIX LIST
+//
+
+
+/////////////////////////////////////
+
+		// HUD! //
+
```
Comment (remove32) [inline fulp_modules/modular_bloodsuckers/bloodsucker/datum_bloodsucker.dm:621]: Uncomment the other shit and add this here:
```
/datum/antagonist/bloodsucker/proc/update_bloodsucker_icons_added(datum/mind/m)
	var/datum/atom_hud/antag/vamphud = GLOB.huds[ANTAG_HUD_BLOODSUCKER]
	vamphud.join_hud(owner.current)
	set_antag_hud(owner.current, "bloodsucker") // "bloodsucker"
	owner.current.hud_list[ANTAG_HUD].icon = image('icons/mob/bloodsucker_icons.dmi', owner.current, "bloodsucker")	// FULP ADDITION! Check prepare_huds in mob.dm to see why.

/datum/antagonist/bloodsucker/proc/update_bloodsucker_icons_removed(datum/mind/m)
	var/datum/atom_hud/antag/vamphud = GLOB.huds[ANTAG_HUD_BLOODSUCKER]
	vamphud.leave_hud(owner.current)
	set_antag_hud(owner.current, null)


/datum/atom_hud/antag/bloodsucker // default antag where we can override behavior below
```

Doing this I was able to get the huds (mostly) working

File: fulp_modules/modular_bloodsuckers/bloodsucker/datum_hunter.dm [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+		to_chat(owner, "<span class='notice'>You've lost the trail.</span>")
+	..()
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/datum/action/bloodsucker/trackvamp/
+	name = "Track Monster"//"Cellular Emporium"
+	desc = "Take a moment to look for clues of any nearby monsters.<br>These creatures are slippery, and often look like the crew."
+	button_icon = 'icons/mob/actions/bloodsucker.dmi'	//This is the file for the BACKGROUND icon
```
Comment (remove32) [inline fulp_modules/modular_bloodsuckers/bloodsucker/datum_hunter.dm:157]: icons/mob/actions/actions_bloodsucker.dmi

File: fulp_modules/modular_bloodsuckers/bloodsucker/datum_hunter.dm [hunk 3]
```diff
@@ -0,0 +1,401 @@
...
+	background_icon_state = "vamp_power_off"		//And this is the state for the background icon
+	icon_icon = 'icons/mob/actions/bloodsucker.dmi'		//This is the file for the ACTION icon
```
Comment (remove32) [inline fulp_modules/modular_bloodsuckers/bloodsucker/datum_hunter.dm:159]: icons/mob/actions/actions_bloodsucker.dmi

**Review by remove32 — CHANGES_REQUESTED** [review]

**Comment by remove32** [comment]
i played around with what you had for a little bit. results are pretty promising, aside from the fact that for whatever reason I kept dropping into hardcrit and dying of oxygen loss. probably to do with lack of heartbeat.

**Comment by JohnFulpWillard** [comment]
> 
> 
> i played around with what you had for a little bit. results are pretty promising, aside from the fact that for whatever reason I kept dropping into hardcrit and dying of oxygen loss. probably to do with lack of heartbeat.

I never had that problem before, bloodsuckers should have trait no_breath which means they cant take oxygen damage, but I'll go back to check again

File: code/datums/hud.dm [hunk 4]
```diff
@@ -28,6 +28,8 @@ GLOBAL_LIST_INIT(huds, list(
 	ANTAG_HUD_FUGITIVE = new/datum/atom_hud/antag(),
 	ANTAG_HUD_GANGSTER = new/datum/atom_hud/antag/hidden(),
 	ANTAG_HUD_SPACECOP = new/datum/atom_hud/antag(),
+	ANTAG_HUD_BLOODSUCKER = new/datum/atom_hud/antag(), // Fulpstation Bloodsucker
```
[the remark this answers is not in the document]
Comment (JohnFulpWillard) [reply code/datums/hud.dm:31]: It used to be like that in previous versions of the draft, but I changed it while refactoring the huds

**Comment by remove32** [comment]
> 
> 
> > i played around with what you had for a little bit. results are pretty promising, aside from the fact that for whatever reason I kept dropping into hardcrit and dying of oxygen loss. probably to do with lack of heartbeat.
> 
> I never had that problem before, bloodsuckers should have trait no_breath which means they cant take oxygen damage (and the trait that stops them from falling in crit), but I'll go back to check again

the trait we need here is TRAIT_STABLEHEART. Part of the wounds update

File: fulp_modules/modular_bloodsuckers/bloodsucker/datum_bloodsucker.dm (hunk 1 above)
Comment (JohnFulpWillard) [reply fulp_modules/modular_bloodsuckers/bloodsucker/datum_bloodsucker.dm:590]: I tried this and I feel like it only broke the huds more, I already added replacements to give said huds using new things /tg/ gave us

**Comment by JohnFulpWillard** [comment]
Closing this, I'll open a new one if I manage to fix the huds

**Pull request closed without merging**